addon/src/main.js

```diff
@@ -43,37 +45,61 @@ const webapp = new WebApp({
   hub
 });
 
-sideEffects.setContext(
-  Object.assign({}, store, {
-    env,
-    feedbackManager,
-    hacks,
-    installManager,
-    loader,
-    notificationManager,
-    self,
-    tabs,
-    telemetry,
-    ui,
-    webapp
-  })
-);
-
 export function main({ loadReason }: { loadReason: string }) {
+  sideEffects.setContext(
+    Object.assign({}, store, {
+      env,
+      feedbackManager,
+      hacks,
+      installManager,
+      loader,
+      notificationManager,
+      self,
+      tabs,
+      telemetry,
+      ui,
+      webapp
+    })
+  );
   env.subscribe(store);
   sideEffects.enable(store);
-  installManager.selfLoaded(loadReason);
-  loader.loadExperiments(
-    startEnv.name,
-    startEnv.baseUrl
-  );
+  loader.loadExperiments(startEnv.name, startEnv.baseUrl);
   notificationManager.schedule();
   feedbackManager.schedule();
   feedbackManager.maybeShare();
+  if (loadReason === 'install' || loadReason === 'enable') {
+    telemetry.setPrefs();
+    telemetry.ping(self.id, 'enabled');
+  }
+  telemetry.sendGAEvent({
+    t: 'event',
+    ec: 'add-on Interactions',
+    ea: 'browser startup',
+    el: Object.keys(activeExperiments(store.getState())).length
+  });
+  Services.obs.addObserver(
+    () => {
+      telemetry.ping(
+        'daily',
+        Object.keys(activeExperiments(store.getState())).length
+      );
+    },
+    'idle-daily',
+    false
+  );
 }
 
 export function onUnload(reason: string) {
-  installManager.selfUnloaded(reason);
+  switch (reason) {
+    case 'uninstall':
+      telemetry.ping(self.id, 'disabled');
```

Comment:
cool. I'm matching the behavior of the old addon [here](https://github.com/mozilla/testpilot/blob/master/old-addon/index.js#L356-L358)